src/pretest.proj

```diff
@@ -2,15 +2,32 @@
   <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.props" />
 
   <PropertyGroup>
-    <!-- Enable GenerateTestSharedFrameworkDepsFile from CoreFxTesting package -->
-    <GenerateTestSharedFrameworkDepsFile Condition="'$(BinplaceTestSharedFramework)' == 'true'">true</GenerateTestSharedFrameworkDepsFile>
+    <BuildDependsOnTargets>FilterProjects;GenerateLaunchSettingsFiles</BuildDependsOnTargets>
+    <BuildDependsOnTargets Condition="'$(BinplaceTestSharedFramework)' == 'true'">$(BuildDependsOnTargets);GenerateTestSharedFrameworkDepsFile</BuildDependsOnTargets>
+    <BuildDependsOnTargets Condition="'$(PlatformManifestFile)' != ''">$(BuildDependsOnTargets);GenerateFileVersionProps</BuildDependsOnTargets>
   </PropertyGroup>
 
   <ItemGroup>
     <!-- This is necessary as inside VS project references that aren't part of the sln aren't built. -->
     <Project Include="$(CommonTestPath)\CoreFx.Private.TestUtilities\CoreFx.Private.TestUtilities.csproj" />
   </ItemGroup>
 
+  <!-- Microsoft.XmlSerializer.Generator should not be marked as a platform item and be copy-local instead. -->
+  <Target Name="CollectSharedFrameworkRuntimeFiles"
+          Condition="'$(PlatformManifestFile)' != ''"
+          BeforeTargets="GenerateFileVersionProps">
+    <ItemGroup Condition="'$(BuildingNETCoreAppVertical)' == 'true'">
+      <_manualSharedFrameworkRuntimeFiles Include="System.Security.Cryptography.Native.OpenSsl.so" />
```

Comment:
Can you just do a glob of the native folder?

Reply:
1) I'm not sure how I would get the file-extensions (so and dylib) and 2) `System.Security.Cryptography.Native.OpenSsl` isn't a file/dir name anywhere.

Reply:
The exact file names will be under $(NativeBinDir).  If they aren't, then I don't imagine you'd care about feeding them to conflict resolution.

Reply:
Just wanted to follow-up on this. `System.Security.Cryptography.Native.OpenSsl` isn't under $(NativeBinDir) but we care about feeding it to conflict resolution as otherwise it would restore runtime assets for it into the test app's output dir.